paasta_tools/firewall.py

```diff
@@ -50,48 +57,95 @@ def chain_name(self):
         return chain
 
     @property
+    def config(self):
+        load_fn = {
+            'chronos': load_chronos_job_config,
+            'marathon': load_marathon_service_config,
+        }[self.framework]
+        return load_fn(
+            self.service, self.instance,
+            load_system_paasta_config().get_cluster(),
+            load_deployments=False,
+            soa_dir=self.soa_dir,
+        )
+
+    @property
     def rules(self):
-        # TODO: actually read these from somewhere
-        return (
-            iptables.Rule(
+        conf = self.config
+
+        rules = [_default_rule(conf)]
+        rules.extend(_well_known_rules(conf))
+        rules.extend(_smartstack_rules(conf, self.soa_dir))
+        return tuple(rules)
+
+    def update_rules(self):
+        iptables.ensure_chain(self.chain_name, self.rules)
+
+
+def _default_rule(conf):
+    policy = conf.get_outbound_firewall()
+    if policy == 'block':
+        return iptables.Rule(
+            protocol='ip',
+            src='0.0.0.0/0.0.0.0',
+            dst='0.0.0.0/0.0.0.0',
+            target='REJECT',
+            matches=(),
+        )
+    elif policy == 'monitor':
+        # TODO: log-prefix
+        return iptables.Rule(
+            protocol='ip',
+            src='0.0.0.0/0.0.0.0',
+            dst='0.0.0.0/0.0.0.0',
+            target='LOG',
+            matches=(),
+        )
+    else:
+        raise AssertionError(policy)
+
+
+def _well_known_rules(conf):
+    for resource in conf.get_dependencies().get('well-known', ()):
+        if resource == 'internet':
+            yield iptables.Rule(
                 protocol='ip',
                 src='0.0.0.0/0.0.0.0',
                 dst='0.0.0.0/0.0.0.0',
-                target='REJECT',
+                target='PAASTA-INTERNET',
                 matches=(),
-            ),
-            iptables.Rule(
-                protocol='tcp',
-                src='0.0.0.0/0.0.0.0',
-                dst='169.254.255.254/255.255.255.255',
-                target='ACCEPT',
-                matches=(
-                    ('tcp', (('dport', '20668'),)),
-                )
-            ),
+            )
+        else:
+            # TODO: handle better
```

Comment:
yea, just log and do nothing. the show must go on here and user typos are not an excuse to crash the whole system.

I was hoping we could make a new log "component" for paasta logs (`security`?) and it could show up there.